blog.md

```diff
@@ -111,38 +166,41 @@ To git@github.com:paulczar/ci-demo.git
 
-![IMAGE JENKINS JOB](http://)
+![IMAGE JENKINS JOB](https://ci-demo-ghost-images.s3.amazonaws.com/2015/Nov/jenkins_pr_testing-1448031335825.png)
 
-Once Jenkins has provided the URL to the epheneral application we can test it there and confirm everything is working fine and you can upload photos etc.
+The ephemeral environment can be viewed at http://pr-xx-ghost.ci-demo.paulczar.net by anyone wishing to review the Pull Request. Subsequent updates to the PR will update the deployed application.
 
-In this case while the blog worked, images could not be uploaded.  This is because it doesn't have any s3 credentials.  I can manually fix this by running `deis config:set -a ghost S3_ACCESS_KEY_ID=*** S3_ACCESS_KEY=***`, if I want this to be a permanent feature I would update the PR builder job in Jenkins to do this as part of the build.
+We can run some manual tests specific to the feature being developed (such as uploading photos) once the URL to the ephemeral application is live.
 
 #### Staging
 
-The `development` branch is protected, and will only accept PRs that have passed tests and had a successful demo environment deployed.
+Jenkins will see that a Pull Request is merged into the development branch and will perform two jobs:
 
-When the `Pull Request` is merged it will fire off two webhooks to Jenkins.  The first will delete the demo application for that PR and the second will update the staging environment in deis (http://stage-ghost.ci-demo.paulczar.net) with the contents of the `development` branch.
+1. Delete the ephemeral environment for Pull Request as it is no longer needed.
+2. Create and deploy a new release of the contents of the development branch to the staging environment in Deis (http://stage-ghost.ci-demo.paulczar.net).
 
-![IMAGE JENKINS JOB](http://)
+![IMAGE JENKINS JOB](https://ci-demo-ghost-images.s3.amazonaws.com/2015/Nov/ci_staging_deploy-1448031350720.png)
 
-Originally when I started building this demo I had assumed that being able to perform actions on PR merges/closes would be simple, but I quickly discovered none of the CI tools support performing actions on PR close. Thankfully I was able to find a useful [blog](http://chloky.com/github-json-payload-in-jenkins/) post that described how to set up a custom job with a webhook that could process the github payload.
+![](https://ci-demo-ghost-images.s3.amazonaws.com/2015/Nov/stage_ghost-1448031723495.png)
 
+_Originally when I started building this demo I had assumed that being able to perform actions on PR merges/closes would be simple, but I quickly discovered that __none of the CI tools that I looked at supported performing actions on PR close__. Thankfully I was able to find a useful [blog](http://chloky.com/github-json-payload-in-jenkins/) post that described how to set up a custom job with a webhook that could process the GitHub payload._
```

Comment:
Careful mixing underscores in the markdown here. This is tricky since different sites use different flavours. Yay computers!